src/libstd/sys/unix/fs.rs

```diff
@@ -498,6 +498,65 @@ impl File {
         Ok(n as u64)
     }
 
+    pub fn set_times(&self,
+                     accessed: SystemTime,
+                     modified: SystemTime,
+                     created: Option<SystemTime>) -> io::Result<()> {
+        if let Some(birthtime) = created {
+            // On some (and maybe all) versions of BSD the birth time can be set
+            // thanks to the guarantee that the birth time is always equal to or
+            // earlier then the last modified time.
+            // So the birth time can be set by setting the times twice:
+            // once with times[1] set to the birth time, and then with times[1]
+            // set to the modified time.
+            try!(self.set_times_inner(accessed, birthtime));
```

Comment:
Out of curiosity, is there precedent for setting the creation time when copying files?